src/main/java/bio/terra/service/common/gcs/GcsUriUtils.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.service.common.gcs;
+
+import bio.terra.service.dataset.exception.InvalidUriException;
+import org.apache.commons.lang3.StringUtils;
+
+public final class GcsUriUtils {
+  public static class GsUrlParts {
+    private String bucket;
+    private String path;
+    private boolean isWildcard = false;
+
+    public String getBucket() {
+      return bucket;
+    }
+
+    public GsUrlParts bucket(String bucket) {
+      this.bucket = bucket;
+      return this;
+    }
+
+    public String getPath() {
+      return path;
+    }
+
+    public GsUrlParts path(String path) {
+      this.path = path;
+      return this;
+    }
+
+    public boolean getIsWildcard() {
+      return isWildcard;
+    }
+
+    public GsUrlParts isWildcard(boolean isWildcard) {
+      this.isWildcard = isWildcard;
+      return this;
+    }
+  }
+
+  /**
+   * Parse a Google Cloud Storage URI into its component pieces
+   *
+   * @param uri of type gs://<bucket_name>/<file_path_inside_bucket>
+   * @return Object representing uri pieces
+   */
+  public static GsUrlParts parseBlobUri(String uri) {
+    String protocol = "gs://";
+    if (!StringUtils.startsWith(uri, protocol)) {
+      throw new InvalidUriException("Ingest source is not a valid gs: URI: '" + uri + "'");
+    }
+    String noGsUri = StringUtils.substring(uri, protocol.length());
+    if (noGsUri.length() < 4) {
+      throw new InvalidUriException("Invalid bucket name in gs path: '" + uri + "'");
+    }
+
+    String bucket = StringUtils.substringBefore(noGsUri, "/");
+
+    if (bucket.indexOf('*') > -1) {
+      throw new InvalidUriException("Bucket wildcards are not supported: URI: '" + uri + "'");
+    }
+
+    String gsBucketPattern = "[a-z0-9_.\\-]{3,222}";
```

Comment:
It's unlikely to be a performance issue, but it seems wasteful to create a new `Pattern` every time this method is called. The usual way to do this is to make a static final Pattern and use `pattern.matcher(s).matches()` in the code.